mass_update_incidents/mass_update_incidents.py

```diff
@@ -43,72 +40,176 @@ def mass_update_incidents(args):
+                # For resolving, we need to calculate total alerts and adjust rate
+                batch_triggered_alerts = sum(incident.get('alert_counts', {}).get('triggered', 0) for incident in batch)
+                print(f"Batch {batch_num + 1}/{num_batches}: Resolving {len(batch)} incidents with {batch_triggered_alerts} total triggered alerts")
+                
+                # Prepare bulk update request for non-skipped incidents
+                incident_updates = list(map(lambda incident: {
+                    "id": incident['id'],
+                    "type": "incident_reference",
+                    "status": "resolved"
+                }, batch))
+                
+                # Send bulk update request and measure time
+                start_time = time.time()
+                session.rput("/incidents", json={
+                    "incidents": incident_updates
+                })
+                processing_time = time.time() - start_time
+                
+                # Calculate minimum time needed based on alert count and rate limit
+                min_time_needed = batch_triggered_alerts / 100.0
+                
+                # Calculate dynamic rate multiplier based on alert count
+                # For small numbers of alerts, use minimal multiplier
+                # For large numbers, apply progressively larger multiplier
+                if batch_triggered_alerts <= ALERT_THRESHOLD:
+                    # For small batches, use a small fixed multiplier (1.5x)
+                    dynamic_multiplier = 1.5
+                else:
+                    # For larger batches, scale up the multiplier based on alert count
+                    # Formula: 1.5 + (alerts - threshold) / threshold 
+                    # Examples: 
+                    # - At threshold of 200: 200 alerts → 1.5x, 400 alerts → 2.5x, 600 alerts → 3.5x
+                    excess_alerts = batch_triggered_alerts - ALERT_THRESHOLD
+                    dynamic_multiplier = 1.5 + (excess_alerts / ALERT_THRESHOLD)
+                    # Cap the multiplier at a reasonable maximum (5.0)
+                    dynamic_multiplier = min(dynamic_multiplier, 5.0)
+                
+                # Apply the dynamic multiplier to account for backend async work
+                adjusted_time_needed = min_time_needed * dynamic_multiplier
+                
+                # Calculate wait time (adjusted time minus the time already spent processing)
+                wait_time = max(1.0, (adjusted_time_needed - processing_time))
+                
+                print(f"  Batch completed in {processing_time:.2f}s, {batch_triggered_alerts} alerts")
+                print(f"  Dynamic rate multiplier: {dynamic_multiplier:.2f}x (based on alert count)")
+                print(f"  Rate limit: {adjusted_time_needed:.2f}s needed")
+                print(f"  Waiting {wait_time:.2f}s before next batch to maintain rate limit")
+                time.sleep(wait_time)
```

Comment:
You shouldn't sleep if the batch is complete (ie. `batch_num+1 == num_batches`)

I ran the script for 3 incidents, and had to wait 133 seconds for the script to complete.

```
Batch 1/1: Resolving 3 incidents with 2697 total triggered alerts
  Batch completed in 1.54s, 2697 alerts
  Dynamic rate multiplier: 5.00x (based on alert count)
  Rate limit: 134.85s needed
  Waiting 133.31s before next batch to maintain rate limit
```